ArduCopter/mode_flip.cpp

```diff
@@ -126,17 +131,24 @@ void ModeFlip::run()
         if (flip_angle >= 4500) {
             if (roll_dir != 0) {
                 // we are rolling
-            _state = FlipState::Roll;
+                _state = FlipState::Roll;
             } else {
                 // we are pitching
                 _state = FlipState::Pitch_A;
-        }
+            }
         }
         break;
 
     case FlipState::Roll:
-        // between 45deg ~ -90deg request 400deg/sec roll
-        attitude_control->input_rate_bf_roll_pitch_yaw(FLIP_ROTATION_RATE * roll_dir, 0.0, 0.0);
+        // keep target aircraft from getting too far away from actual aircraft
+        if (flip_angle_error > 1500 && flip_angle_error <= 4500) {
+            float knock_down = 1.0f - ((float)flip_angle_error - 1500.0f) / 3000.0f;
+            flip_rate_cdps = (ahrs.get_gyro().x + knock_down * (flip_rate_cdps - ahrs.get_gyro().x * 5730)) * roll_dir;
```

Comment:
```suggestion
            flip_rate_cdps = (ahrs.get_gyro().x + knock_down * (flip_rate_cdps - degrees(ahrs.get_gyro().x) * 100)) * roll_dir;
```
The `degrees` function makes it a little easier to follow.